interpreter/interpreter.go

```diff
@@ -313,10 +310,22 @@ func (itrp *Interpreter) doExpression(expr semantic.Expression, scope Scope) (va
 			return nil, err
 		}
 
+		ltyp := itrp.typeof(e.Left, l.Type())
+		rtyp := itrp.typeof(e.Right, r.Type())
+		// TODO(jsternberg): This next section needs to be removed
+		// since type inference should give the correct type.
+		if ltyp == semantic.Nil && l.Type() != semantic.Nil {
+			// There's a weird bug in type inference where it
+			// determines the type is null even when it's not.
```

Comment:
We talked in Slack, but I'd like to see a test case added to the semantic package highlighting this bug. This appears to be an existing issue, but I want to make sure that the recent null work hasn't expanded the bug.